hawkular/alerts.py

```diff
@@ -0,0 +1,219 @@
+"""
+   Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+   and other contributors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+
+from client import ApiOject, HawkularBaseClient
+
+
+class Trigger(ApiOject):
+    def __init__(self, dictionary=dict()):
+        udict = Trigger.transform_dict_to_underscore(dictionary)
+        self.id = udict.get('id')
```

Comment:
@rubenvp8510 this seems a lot of manual work. Can't you just leverage [`__slots__`](https://docs.python.org/2/reference/datamodel.html#slots):

```python
class Trigger(ApiOject):
    __slots__ = [
        name,
        description,
        ...
    ]

    def __init__(self, dictionary=dict()):
        udict = Trigger.transform_dict_to_underscore(dictionary)
        for k in self.__slots__:
            setattr(self, k, udict.get(k))
```

This seems much simpler and less error-prone.